src/mongocxx/test/client_session.cpp

```diff
@@ -60,6 +64,35 @@ TEST_CASE("session options", "[session]") {
         auto s = c.start_session(opts);
         REQUIRE(!s.options().causal_consistency());
     }
+
+    SECTION("set snapshot consistency") {
+        options::client_session opts;
+
+        REQUIRE_FALSE(opts.snapshot());
+        opts.causal_consistency(false);
+        opts.snapshot(true);
+        REQUIRE(opts.snapshot());
+        REQUIRE_FALSE(opts.causal_consistency());
+
+        auto s = c.start_session(opts);
+
+        REQUIRE(opts.snapshot());
+        REQUIRE_FALSE(opts.causal_consistency());
+    }
+
+    SECTION("causal and snapshot consistency are mutually exclusive") {
+        // Trying to set both casusal and snapshot consistency should result
+        // in an error when starting the session:
+        options::client_session opts;
+
+        opts.snapshot(true);
+        opts.causal_consistency(true);
+
+        REQUIRE(opts.causal_consistency());
+        REQUIRE(opts.snapshot());
+
+        REQUIRE_THROWS_AS(c.start_session(opts), mongocxx::exception);
+    }
 }
 
```

Comment:
In addition to the new prose test, the specification change in [DRIVERS-1607](https://jira.mongodb.org/browse/DRIVERS-1607) includes session tests in the Unified Test Format: https://github.com/mongodb/specifications/tree/master/source/sessions/tests/unified. I filed CXX-2371 as part of this epic to add these tests and update the unified test runner.